From 7b1468de8681994b56fbbcaa951e504ab96b8eb0 Mon Sep 17 00:00:00 2001
From: Dave Young <dyoung@redhat.com>
Date: Thu, 22 May 2014 02:43:08 -0400
Subject: [firmware] efi: Remove unused variables in __map_region()

Message-id: <1400726607-13717-17-git-send-email-dyoung@redhat.com>
Patchwork-id: 80117
O-Subject: [RHEL7.1 PATCH 16/35] x86/efi: Remove unused variables in __map_region()
Bugzilla: 1080109
RH-Acked-by: Chao WANG <chaowang@redhat.com>
RH-Acked-by: Vivek Goyal <vgoyal@redhat.com>

Resolves: bz1080109

Brew build:
https://brewweb.devel.redhat.com/taskinfo?taskID=7480539

Test: Tested kexec/kdump on HP workstation and qemu.

Upstream commit:
commit 2da6e57cce14a1c3b0692d6f877b72e185110e2e
Author: Dave Young <dyoung@redhat.com>
Date:   Fri Dec 20 18:02:13 2013 +0800

    x86/efi: Remove unused variables in __map_region()

    variables size and end is useless in this function, thus remove them.

    Reported-by: Toshi Kani <toshi.kani@hp.com>
    Tested-by: Toshi Kani <toshi.kani@hp.com>
    Signed-off-by: Dave Young <dyoung@redhat.com>
    Acked-by: Borislav Petkov <bp@suse.de>
    Signed-off-by: Matt Fleming <matt.fleming@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/platform/efi/efi_64.c b/arch/x86/platform/efi/efi_64.c
index bf286c3..c5a6491 100644
--- a/arch/x86/platform/efi/efi_64.c
+++ b/arch/x86/platform/efi/efi_64.c
@@ -148,15 +148,11 @@ void efi_setup_page_tables(void)
 static void __init __map_region(efi_memory_desc_t *md, u64 va)
 {
  pgd_t *pgd = (pgd_t *)__va(real_mode_header->trampoline_pgd);
- unsigned long pf = 0, size;
- u64 end;
+ unsigned long pf = 0;
 
  if (!(md->attribute & EFI_MEMORY_WB))
   pf |= _PAGE_PCD;
 
- size = md->num_pages << PAGE_SHIFT;
- end  = va + size;
-
  if (kernel_map_pages_in_pgd(pgd, md->phys_addr, va, md->num_pages, pf))
   pr_warn("Error mapping PA 0x%llx -> VA 0x%llx!\n",
       md->phys_addr, va);
-- 
1.7.1